rust/src/core.rs

```diff
@@ -38,6 +39,36 @@ pub const STREAM_TOCLIENT: u8 = 0x08;
 pub const STREAM_GAP:      u8 = 0x10;
 pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
+pub const DIR_TOSERVER:    u8 = 0b0000_0100;
+pub const DIR_TOCLIENT:    u8 = 0b0000_1000;
+
+#[repr(C)]
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Direction {
+    ToServer = 0x04,
+    ToClient = 0x08,
+}
+
+impl Default for Direction {
+    fn default() -> Self { Direction::ToServer }
+}
+
+impl From<u8> for Direction {
+    fn from(d: u8) -> Self {
+        if d == DIR_TOSERVER {
+            Direction::ToServer
+        } else if d == DIR_TOCLIENT {
+            Direction::ToClient
+        } else if d & DIR_TOSERVER != 0 {
+            Direction::ToServer
+        } else if d & DIR_TOCLIENT != 0 {
+            Direction::ToClient
+        } else {
+            debug_validate_fail!("Unknown direction!!");
```

Comment:
Should we `debug_validate_fail ` if we have both `DIR_TOSERVER` and `DIR_TOCLIENT ` set ?

Reply:
hmm makes sense to me. @victorjulien ?

Reply:
> hmm makes sense to me. @victorjulien ?

Makes sense to me as well.